### ISSUE TITLE
New value type (array) is not in annotated types.

### ISSUE THREAD

**Issue by cpriest** [kalessil/phpinspectionsea#469]
## Sample Code
```php
	/**
	 * @return string|string[]
	 */
	function foo() {
		if(mt_rand(0,1) == 1)
			return ['', ''];
		return '';
	}

	/**
	 * @param string $lorem
	 */
	function bar($lorem) {
		$lorem = foo();
	}
```
### Expected
No inspection error.

### Actual
New value type (array) is not in annotated types on `$lorem = foo();`

From Architecture >> Callable parameter usage violates definition

## Comments
I believe this is a bug, foo() returns either a `string` or a `string[]` and `$lorem` is declared as a `string`.

**Comment by kalessil** [comment]
Look like a correct reporting: `$lorem` is expected to be a string only, but can store array.

If the bar PhpDoc is correct, then a solution would look like:
```php
$value  = foo();
$lorem =  is_array($value) ? $value[...] : $value;
```

**Comment by kalessil** [comment]
I'll close the issues, but we can continue the discussion here.

**Comment by rentalhost** [comment]
A possible fix is change the param type to `string|string[]`, or create a new variable that can handle a new type collection.
